### PR TITLE
chore(oauth): Added method signatures as per JSDoc conventions

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Follow up of #485 #476

### CODE REVIEW

File: assets/js/oauth.js [hunk 1]
```diff
@@ -24,17 +30,30 @@ const sendPostRequest = async (url, params) => {
...
+ */
+
+const parseQueryString = searchQuery => {
+  if (searchQuery === "") {
     return {};
   }
-  const segments = string.split("&").map(s => s.split("="));
+  const segments = searchQuery.split("&").map(s => s.split("="));
   const queryString = {};
   segments.forEach(s => (queryString[s[0]] = s[1]));
```
Comment (jamesgeorge007) [inline assets/js/oauth.js]:
```js
  const queryString = segments.reduce(
    (obj, el) => ({ ...obj, [el[0]]: el[1] }),
    {}
  );
```

Comment (jamesgeorge007) [reply assets/js/oauth.js]: @liyasthomas what do you think about the above approach?

Comment (jamesgeorge007) [reply assets/js/oauth.js]: Does it make the code hard to read:question:

Comment (liyasthomas) [reply assets/js/oauth.js]: I thought `forEach` made more sense but going through `reduce` implementation, I think devs prefer `reduce` over `forEach` to make new object. So I :+1:  that.

**Comment by liyasthomas** [comment]
Do not merge, there is an error.

**Pull request merged**